docs/user-guide/petset.md

```diff
@@ -0,0 +1,269 @@
+* Node: A single virtual or physical machine in a Kubernetes cluster.
+* Cluster: A group of nodes in a single failure domain, unless mentioned otherwise.
+* Persistent Volume Claim: A request for storage, typically a [persistent volume](/docs/user-guide/persistent-volumes/walkthrough.md).
+* Host name: The hostname attached to the UTS namespace of the pod, i.e the output of `hostname` in the pod.
+* DNS/Domain name: A *cluster local* domain name resolvable using standard methods (eg: [gethostbyname](http://linux.die.net/man/3/gethostbyname)).
+* Ordinality: the proprety of being "ordinal", or occupying a position in a sequence.
+* Pet: a single member of a Pet Set; more generally, a stateful application.
+* Peer: a process running a server, capable of communicating with other such processes.
+
+__Prerequisites__
+
+This doc assumes familiarity with the following Kubernetes concepts:
+
+* [Pods](/docs/user-guide/gc/pods/single-container.md)
+* [Cluster DNS](/docs/admin/dns.md)
+* [Headless Services](/docs/user-guide/gc/services/#headless-services)
+* [Persistent Volumes](/docs/user-guide/gc/volumes.md)
+* [Dynamic volume provisioning](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md)
+
+You need a working Kubernetes cluster at version >= 1.3, with a healthy DNS [cluster addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/README.md) at version >= 15. You cannot use Pet Set on a hosted Kubernetes provider that has disabled `alpha` resources.
+
+## What is a Pet Set?
+
+In Kubernetes, most pod management abstractions group them into disposable units of work that compose a micro service. Replication controllers for example, are designed with a weak guarantee - that there should be N replicas of a particular pod template. The pods are treated as stateless units, if one of them is unhealthy or superseded by a better version, the system just disposes it.
+
+```
+    foo.default.svc.cluster.local
+             |service|
+             /       \
+    | pod-asdf |    | pod-zxcv |
+```
+
+A Pet Set in contrast, is a group of stateful pods that form a clustered application.
+
+```
+   *.foo.default.svc.cluster.local
+    | mysql-0 | <-> | mysql-1 |
+      [pd 0]          [pd 1]
+```
+
+The co-ordinated deployment of clustered applications is notoriously hard. They require stronger notions of identity and membership, which they use in opaque internal protocols, and are especially prone to race conditions and deadlock. Traditionally administrators have deployed these applications by leveraging nodes as stable, long-lived entities with persistent storage and static ips.
+
+The goal of Pet Set is to decouple this dependency by assigning identities to individual instances of an application that are not anchored to the underlying physical infrastructure. For the rest of this document we will refer to these entities as "Pets". Our use of this term is predated by the "Pets vs Cattle" analogy.
```

Comment:
Are "Pets" Pods?  What is the relationship between Pet Set and a "Pet"?  e.g. Pet Sets Manage the creation, deletion, recreation on failure, etc?